decidim-bulletin_board-app/app/assets/javascripts/sandbox/identification_keys.es6

```diff
@@ -0,0 +1,267 @@
+
+    async upload(event, alwaysMatches = false) {
+      if (!this.browserSupport || this.privateKey !== null) {
+        return false;
+      }
+
+      return new Promise((resolve, reject) => {
+        if (event) {
+          this.onUploadInputChange(event, resolve, reject, alwaysMatches);
+        } else {
+          let element = document.createElement("input");
+          element.setAttribute("type", "file");
+          element.setAttribute("accept", ".jwk");
+          element.style.display = "none";
+          document.body.appendChild(element);
+
+          element.addEventListener("change", (event) => {
+            document.body.removeChild(element);
+            this.onUploadInputChange(event, resolve, reject, alwaysMatches);
+          });
+          element.click();
+        }
+      });
+    }
+
+    onUploadInputChange(event, resolve, reject, alwaysMatches) {
+      const reader = new FileReader();
+      reader.readAsText(event.target.files[0]);
+      reader.onload = (readEvent) => {
+        let jwk = "";
+        try {
+          jwk = JSON.parse(readEvent.target.result);
+        } catch (error) {
+          return reject("invalid_format");
+        }
+
+        return this.crypto.subtle
+          .importKey(this.format, jwk, this.algorithm, false, this.usages)
+          .then((privateKey) => {
+            const uploadedPublicKey = this._publicKeyFromPrivateKey(jwk);
+            if (
+              alwaysMatches ||
+              this._matchesStoredPublicKey(uploadedPublicKey)
+            ) {
+              this.publicKey = uploadedPublicKey;
+              this.privateKey = privateKey;
+              this._save()
+                .then(() => resolve(true))
+                .catch(() => {
+                  () => reject("could not be saved");
+                });
+            } else {
+              reject("invalid_public_key");
+            }
+          })
+          .catch(() => {
+            reject("invalid_key");
+          });
+      };
+    }
+
+    reset() {
+      this.privateKey = this.publicKey = null;
+      return this._clear();
+    }
+
+    async sign(payload) {
+      if (!this.browserSupport || this.privateKey === null) {
+        return false;
+      }
+
+      const data = `${this.jwtHeader}.${this._encode64(
+        JSON.stringify(payload)
+      )}`;
+      const signature = await this.crypto.subtle.sign(
+        this.algorithm,
+        this.privateKey,
+        this.textEncoder.encode(data)
+      );
+      return `${data}.${btoa(
+        Reflect.apply(String.fromCharCode, null, new Uint8Array(signature))
+      )
+        .replace(/[=]/g, "")
+        .replace(/\+/g, "-")
+        .replace(/\//g, "_")}`;
+    }
+
+    _checkBrowserSupport() {
+      this.indexedDB =
+        window.indexedDB ||
+        window.mozIndexedDB ||
+        window.webkitIndexedDB ||
+        window.msIndexedDB;
+      this.crypto = window.crypto || window.msCrypto;
+      return window.indexedDB && window.crypto;
```

Comment:
Should this be:
```suggestion
      return this.indexedDB && this.crypto;
```

Reply:
Hey @agustibr ! The `identification_keys` file has been copied from the one that we have on `decidim` and I didn't want to change it that much because I am planning to move it to our library as soon as possible. Then I will apply your suggested change, ok? 😄